### ISSUE TITLE
[Bug]: ChoiceFilterState produces invalid call for date and datetime

### ISSUE THREAD

**Issue by gogonzo** [insightsengineering/teal.slice#215]
### What happened?

FilterState is initialized in the specific way. Normally R6 class follows the base class, so that numeric -> RangeFilterState, date -> DateFilterState, .... But, If there are less than `getOption("teal.threshold_slider_vs_checkboxgroup")` unique values, then ChoicesFilterState is initialized. Please analyse the example below, ADSL has extra columns (`numeric_choices`, `date_choices`, `datetime_choices`) and these filters are initialized in the filter-panel.
The call which is produced for all is `<column name> %in% c(<selected character>)`, which means that it looks like this:

- numeric_choices %in% c("1", "2", "3") ✅ 
- date_choices %in% c("2022-02-23", "2022-02-24", "2022-02-25") ❌ 
- datetime_choices %in% c("2022-02-23 22:22:22", "2022-02-24 10:22:22", "2022-02-24 22:22::22") ❌ 

`get_call` for date_choices datetime_choices produces a wrong condition which is FALSE for anything selected. 

```r
options(teal.log_level = "TRACE", teal.show_js_log = TRUE)
# options("teal.bs_theme" = bslib::bs_theme(version = 4))
# options(shiny.trace = TRUE)
devtools::load_all("teal.slice")
library(teal)
library(scda)

dates <- as.Date("2022-02-22") + 1:4
datetimes <- as.POSIXct("2022-02-22 22:22:22") + 1:4 * 43200

ADSL <- synthetic_cdisc_data("latest")$adsl
ADSL$numeric_choices <- sample(1:4, nrow(ADSL), replace = TRUE)
ADSL$date_choices <- sample(dates, nrow(ADSL), replace = TRUE)
ADSL$datetime_choices <- sample(datetimes, nrow(ADSL), replace = TRUE)

ADTTE <- synthetic_cdisc_data("latest")$adtte
ADRS <- synthetic_cdisc_data("latest")$adrs

funny_module <- function(label = "Filter states", datanames = "all") {
  checkmate::assert_string(label)
  module(
    label = label,
    filters = datanames,
    ui = function(id, ...) {
      ns <- NS(id)
      div(
        h2("The following filter calls are generated:"),
        verbatimTextOutput(ns("filter_states")),
        verbatimTextOutput(ns("filter_calls")),
        actionButton(ns("reset"), "reset_to_default")
      )
    },
    server = function(input, output, session, data, filter_panel_api) {
      checkmate::assert_class(data, "tdata")
      observeEvent(input$reset, set_filter_state(filter_panel_api, default_filters))
      output$filter_states <- renderText({
        logger::log_trace("rendering text1")
        filter_panel_api |>
          get_filter_state() |>
          yaml::as.yaml()
      })
      output$filter_calls <- renderText({
        logger::log_trace("rendering text2")
        attr(data, "code")()
      })
    }
  )
}

default_filters <- list(
  ADSL = list(
    numeric_choices = list(selected = 1:3),
    date_choices = list(selected = dates[1:3]),
    datetime_choices = list(selected = datetimes[1:3])
  )
)


data <- cdisc_data(
  cdisc_dataset("ADSL", ADSL),
  cdisc_dataset("ADTTE", ADTTE),
  cdisc_dataset("ADRS", ADRS)
)

app <- init(
  data = data,
  modules = list(funny_module(), funny_module("module2")),
  filter = default_filters
)

runApp(app)
```

**Comment by chlebowa** [comment]
That's because the selection is a character vector and the variables are not. `numeric_choices %in% c("1", "2", "3")` only works because the input to `ChoicesFilterState` is coerced to character before choices are even presented.

**Comment by chlebowa** [comment]
In the wild (on `main`) dates and datetimes do not call `ChoicesFilterState` regardless of number of available values.

**Comment by chlebowa** [comment]
Either fixed by #193 or I cannot reproduce, or I don't entirely understand what is expected.
